File: JS/core/listex.js
import { ofArray, singleton } from "../.fable/fable-library.3.0.0/List.js";

export function call(f, lst) {
    if (lst.tail == null) {
    }
    else {
        const x = lst;
        f(x);
    }
}

export function execute(f, defaultValue, lst) {
    if (lst.tail == null) {
        return defaultValue;
    }
    else {
        const x = lst;
        return f(x);
    }
}

export function of1(a) {
    return singleton(a);
}

export function of2(a, b) {
    return ofArray([a, b]);
}

export function of3(a, b, c) {
    return ofArray([a, b, c]);
}

export function of4(a, b, c, d) {
    return ofArray([a, b, c, d]);
}

export function of5(a, b, c, d, e) {
    return ofArray([a, b, c, d, e]);
}

File: src/core/listex.fs
module List

let call f lst =
  match lst with
  | [] -> ()
  | x -> f x

let execute f defaultValue lst =
  match lst with
  | [] -> defaultValue
  | x -> f x

let of1 a = [ a ]
let of2 a b = [ a; b ]
let of3 a b c = [ a; b; c ]
let of4 a b c d = [ a; b; c; d ]
let of5 a b c d e = [ a; b; c; d; e ]
